### PR TITLE
Ft user profile

### PR DESCRIPTION
**Pull request by omudavid**
Updated UI Form Errors

### CODE REVIEW

File: app/src/main/java/com/decagonhq/decapay/feature/logexpense/presentation/LogExpenseBottomSheetFragment.kt [hunk 1]
```diff
@@ -129,6 +130,28 @@ class LogExpenseBottomSheetFragment : BottomSheetDialogFragment() {
...
+            binding.logExpenseBottomSheetFragmentDescriptionTil.error = ""
+        }
+        if(binding.logExpenseBottomSheetFragmentTransactionDateTv.text.trim().toString().isEmpty()){
+            binding.logExpenseBottomSheetFragmentDateTil.error = "Transaction date is required"
+        }else{
+
+            binding.logExpenseBottomSheetFragmentDateTil.error = ""
+        }
+    }
+
```
Comment (olalekan-oloba) [inline app/src/main/java/com/decagonhq/decapay/feature/logexpense/presentation/LogExpenseBottomSheetFragment.kt:154]: violate DRY. this is repetition

File: app/src/main/java/com/decagonhq/decapay/feature/createbudget/presentation/CreateBudgetFragment.kt [hunk 2]
```diff
@@ -265,6 +270,35 @@ class CreateBudgetFragment : Fragment() {
...
+        if (binding.createBudgetFragmentDescriptionTiedt.text?.trim().toString().isEmpty()) {
+            binding.createBudgetFragmentPeriodTil.error = "Budget period is required"
+        }else{
+           binding.createBudgetFragmentPeriodTil.error = ""
+        }
+        if (binding.createBudgetFragmentBudgetPeriodWeeklyDurationEdittext.text.trim().toString()
+                .isEmpty()
+        ) {
+
+            binding.createBudgetFragmentDescriptionTil.error = "Budget description is required"
```
Comment (Osehi) [inline app/src/main/java/com/decagonhq/decapay/feature/createbudget/presentation/CreateBudgetFragment.kt]: Hello David,
Please, note that "Budget Description" is not a required field.
It is nullable.
Hence, no need for the error flag -
Screenshot 2022-09-26 at 09 22 40
 "Budget description is required".

**Review by Osehi — COMMENTED** [review]
Hello David,
Please, note that "Budget Description" is not a required field.
It is nullable.
Hence, no need for the error flag -
Screenshot 2022-09-26 at 09 22 40
 "Budget description is required".

**Review by Osehi — APPROVED** [review]
Good

File: app/src/main/java/com/decagonhq/decapay/feature/editlogexpense/presentation/EditLogExpenseBottomSheetFragment.kt [hunk 3]
```diff
@@ -128,6 +129,29 @@ class EditLogExpenseBottomSheetFragment : BottomSheetDialogFragment() {
...
+        }
+
+        if(binding.editLogExpenseeBottomSheetFragmentTransactionDateTv.text.trim().toString().isEmpty()){
+            binding.editLogExpenseeBottomSheetFragmentDateTil.error = "Transaction date is required"
+        }else{
+
+            binding.editLogExpenseeBottomSheetFragmentDateTil.error = ""
+        }
+    }
+
```
Comment (olalekan-oloba) [inline app/src/main/java/com/decagonhq/decapay/feature/editlogexpense/presentation/EditLogExpenseBottomSheetFragment.kt]: violate DRY principle

**Review by olalekan-oloba — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/decagonhq/decapay/feature/createbudget/presentation/CreateBudgetFragment.kt (hunk 2 above)
Comment (olalekan-oloba) [reply app/src/main/java/com/decagonhq/decapay/feature/createbudget/presentation/CreateBudgetFragment.kt]: Budget description is not required

**Review by olalekan-oloba — APPROVED** [review]

**Pull request merged**